x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1890,16 +1898,43 @@ public BiConsumer<DiscoveryNode, ClusterState> getJoinValidator() {
     @Override
     public void reload(Settings settings) throws Exception {
         if (enabled) {
-            realms.get().stream().filter(r -> JwtRealmSettings.TYPE.equals(r.realmRef().getType())).forEach(realm -> {
-                if (realm instanceof JwtRealm jwtRealm) {
-                    jwtRealm.rotateClientSecret(
-                        CLIENT_AUTHENTICATION_SHARED_SECRET.getConcreteSettingForNamespace(realm.realmRef().getName()).get(settings)
-                    );
-                }
-            });
+            final List<Exception> exceptions = new ArrayList<>();
+            try {
+                reloadRemoteClusterCredentials(settings);
+            } catch (Exception ex) {
+                exceptions.add(ex);
+            }
+
+            try {
+                reloadSharedSecretsForJwtRealms(settings);
+            } catch (Exception ex) {
+                exceptions.add(ex);
+            }
+
+            ExceptionsHelper.rethrowAndSuppress(exceptions);
+        } else {
+            ensureNoRemoteClusterCredentialsOnDisabledSecurity(settings);
         }
     }
 
+    private void reloadSharedSecretsForJwtRealms(Settings settingsWithKeystore) {
+        realms.get().stream().filter(r -> JwtRealmSettings.TYPE.equals(r.realmRef().getType())).forEach(realm -> {
+            if (realm instanceof JwtRealm jwtRealm) {
+                jwtRealm.rotateClientSecret(
+                    CLIENT_AUTHENTICATION_SHARED_SECRET.getConcreteSettingForNamespace(realm.realmRef().getName()).get(settingsWithKeystore)
+                );
+            }
+        });
+    }
+
+    private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
+        // Accepting a blocking call here since the underlying action is local-only and only performs fast in-memory ops
```

Comment:
When I add connection re-building in a follow up we will need to revisit this. Connections can take some time to re-build so blocking on the transport worker thread is less viable. I'd like to tackle this in the next PR though since the changes are quite big already.